keps/sig-release/1731-publishing-packages/README.md

```diff
@@ -171,7 +179,6 @@ A configuration for the package managers might look something like:
 
 Different architectures will be published into the same repos, it is up to the package managers to pull and install the correct package for the target platform.
 
-
 Ideally, publishing/promoting a package means to commit a change to a configuration file which triggers a "package promotion tool".
 That tool ...
 - manages which packages need to go into which `${channel}` for which `${dist}` of which `${k8s_release}`
```

Comment:
Ok. Need some clarification here, when we talk about "file promotion" what do we mean? Is that promotion of packages through the channels from nightly, to dev to stable channels? Or are we talking about the package contents so that end users can see what files have changed in a package?

Reply:
With file promotion I'm referring to this effort: https://github.com/kubernetes-sigs/promo-tools/issues/413 

We do not use this process yet but plan to for the binary artifacts later on. I don't think we have to wire this effort into the deb/rpm story yet to keep things separated.

Reply:
OK, I disagree with both using and not using file promotion. Publishing the packages is a different beast altogether, ie we cannot use a simple copy or file promotion. 

There is a process that needs to run after forging a new set of packages both in .rpm (`updaterepo`) and .deb (`dpkg-scanpackages`). the new packages need to land in their repositories and then the respective tool needs to be run to update the repo's index.

If we do this as part of a promotion process (and I could do a strong argument about favoring this design), we could add the index update to kpromo (the "image promoter").

On the other hand we cannot "do nothing". The process needs to run to update the indices to that OS tools (apt/yum/dnf) can pick up updates.

I don't think the KEP would be complete without defining how we plan to do this. The end-user story is not complete without finishing this part.

Reply:
@detiber Re:

> If we end up using one or more third party provider(s) for hosting deb/rpm packages

I think we should still dump ideas on the enhancement discussion to plan on building our own repo. I'm not against using a third party by any means, but I think if we choose to go that route we will need to write another KEP. This one is specific to publishing on community infra. WDYT?

Reply:
cc @justaugustus

Reply:
@puerco my though was before we go down the path of spinning up community run infrastructure and taking on the complications related to that, especially around gpg key management, we could do an initial spike using third party hosting first, especially since it could potentially save us a lot of implementation, on going maintenance, and security related toil.

That way, in the event that third party hosting does not work for us, we could then start getting into the detailed implementation details around what is needed, and we could update the KEP with the more detailed plan.

Reply:
I'll leave most of the details out for now and make the promotion optional (since it's based on a different story).